### PR TITLE
CacheChooserController robustness

### PR DESCRIPTION
**Pull request by ScoreUnder**

### CODE REVIEW

File: src/main/kotlin/controllers/CacheChooserController.kt [hunk 1]
```diff
@@ -68,32 +69,34 @@ class CacheChooserController(
             isVisible = false
             selectionMode = ListSelectionModel.SINGLE_SELECTION
             addListSelectionListener {
-                if (selectedIndex != -1) {
-                    btnDownload.isEnabled = true
-                }
+                btnDownload.isEnabled = selectedIndex != -1
```
Comment (ConnorDY) [inline src/main/kotlin/controllers/CacheChooserController.kt:72]: I would add a `CHANGELOG.md` entry under "Bug Fixes" for this.

File: src/main/kotlin/controllers/CacheChooserController.kt [hunk 2]
```diff
@@ -68,32 +69,34 @@ class CacheChooserController(
...
+                downloadCache(it, { path ->
                     lblStatusText.text = ""
                     txtCacheLocation.text = path
                     btnDownload.isEnabled = true
-                }
+                }, { err ->
+                    lblStatusText.text = ""
+                    lblErrorText.text = "Failed to download cache: $err"
+                    btnDownload.isEnabled = true
+                })
```
Comment (ConnorDY) [inline src/main/kotlin/controllers/CacheChooserController.kt:97]: Does this address my comment on this issue:
https://github.com/ConnorDY/OSRS-Environment-Exporter/issues/57#issuecomment-1212690497

**Review by ConnorDY — CHANGES_REQUESTED** [review]
As part of this PR, could you also address my comment on this issue:
https://github.com/ConnorDY/OSRS-Environment-Exporter/issues/57#issuecomment-1212690497

If so, let's add an entry to `CHANGELOG.md` for this and close out that ticket.

File: src/main/kotlin/controllers/CacheChooserController.kt [hunk 3]
```diff
@@ -209,6 +215,9 @@ class CacheChooserController(
 
         populateCachesList(cacheListModel, listCaches, listCachesPlaceholder)
         txtCacheLocation.text = configOptions.lastCacheDir.value.get()
+        if (txtCacheLocation.text.isEmpty()) {
+            btnLaunch.isEnabled = false
+        }
```
Comment (ConnorDY) [inline src/main/kotlin/controllers/CacheChooserController.kt:220]: I would add a `CHANGELOG.md` entry under "Bug Fixes" for this.

**Review by ConnorDY — CHANGES_REQUESTED** [review]

**Comment by ScoreUnder** [comment]
I've suppressed the stack trace for the JSON parse failure (test it out by putting a random character into your xteas json lol) but otherwise not sure what you mean with the error message thing, it seems like it already did that?

**Review by ConnorDY — APPROVED** [review]

**Pull request merged**